quiz-microsevice/data-access/repositories/client-repository.js

```diff
@@ -12,8 +14,34 @@ const createOne = async (name, clientId, clientSecretHash) => {
 
 const retrieveOneByClientId = async (clientId) => {
   const client = await Client.findOne({ clientId });
-
+  
   return client ? toDTO(client) : null;
 };
 
-module.exports = { createOne, retrieveOneByClientId };
+const retrieveAllClients = async () => {
+  
+  const clients = await Client.find().select('-clientSecretHash');
+  
+  return clients.length > 0 ? clients : [];
```

Comment:
Use Array.map() to map all clients to DTO, You don't need to check array length return it any way.